File: signal-matrix/javascript/signals/ai_unknown.js
'use strict';
/**
 * UNKNOWN: AI — module required, only capabilities metadata returned.
 * LLM calls exist in this module but are never invoked from server.js.
 */
const axios = require('axios');

function getCapabilities() {
    return { model: 'gpt-4', maxTokens: 4096 }; // safe — no LLM call
}

async function runUncheckedLlmUnknown(userPrompt) {
    // LLM01 UNKNOWN: user input to LLM, module required but never called
    const resp = await axios.post('https://api.openai.com/v1/chat/completions', {
        model: 'gpt-4',
        messages: [{ role: 'user', content: userPrompt }]
    });
    return resp.data.choices[0].message.content;
}

function evalLlmOutputUnknown(llmCode) {
    // LLM05 UNKNOWN: eval(LLM output), never called
    return eval(llmCode);
}

module.exports = { getCapabilities, runUncheckedLlmUnknown, evalLlmOutputUnknown };

File: signal-matrix/python/signals/ai_unknown.py
# Copyright © 2026 Sthenos Security. All rights reserved.
"""UNKNOWN: AI — module imported, safe function called, vulnerable LLM call never reached."""

def get_model_info() -> dict:
    """Safe — returns model metadata only. Called from entrypoint."""
    return {"model": "gpt-4", "provider": "openai", "version": "1.0"}

def run_unchecked_llm_unknown(user_prompt: str) -> str:
    """
    AI UNKNOWN: Direct user input to LLM without sanitization (LLM01 Prompt Injection).
    Module IS imported but this function is NEVER called from any entrypoint.
    Expected: UNKNOWN (import graph hit but no call path).
    """
    from openai import OpenAI
    client = OpenAI()
    # VIOLATION: unsanitized user_prompt directly in messages
    resp = client.chat.completions.create(
        model="gpt-4",
        messages=[{"role": "user", "content": user_prompt}]
    )
    return resp.choices[0].message.content

def execute_llm_output_unknown(llm_response: str) -> None:
    """
    AI UNKNOWN: LLM05 — eval(LLM output). Never called.
    """
    eval(llm_response)  # VIOLATION: executing untrusted LLM output
